improver_tests/generate_ancillaries/test_GenerateTimezoneMask.py

```diff
@@ -286,69 +312,97 @@ def test__group_timezones_empty_group(timezone_mask):
 def process_expected_fixture() -> callable:
     """Returns expected results for parameterized process tests."""
 
-    def _make_expected(time, grid) -> dict:
+    def _make_expected(grouping, time, grid) -> dict:
```

Comment:
I like the phrase "gloriously complicated". I have made the whole thing into a global dictionary. I have kept factored out only the components that repeat, everything else is now set within the dictionary.

I've kept it next to the tests relating to process rather than putting it at the top of the file (as globals should be). Not sure if you prefer I move it? I think it is more useful to have near the tests where it is used.